views.go

```diff
@@ -1,22 +1,74 @@
 package main
 
 import (
+	"bytes"
 	"database/sql"
 	"fmt"
 	"github.com/labstack/echo"
+	"io/ioutil"
 	"net/http"
 	"net/url"
 	"strings"
 )
 
+func RegisterGET(c echo.Context) error {
+
+	data := NewTemplateGlobal()
+
+	service := ValidateService(c)
+	if service == nil {
+		return c.Render(http.StatusUnauthorized, "unauthorized", data)
+	}
+	data.CASRegister = GetCasRegisterUrl(service.String())
+	data.CASLogin = GetCasLoginUrl(service.String())
+	return c.Render(http.StatusOK, "register", data)
+}
+
+func RegisterPOST(c echo.Context) error {
+
+	data := NewTemplateGlobal()
+
+	service := ValidateService(c)
+	if service == nil {
+		return c.Render(http.StatusUnauthorized, "unauthorized", data)
+	}
+	data.CASRegister = GetCasRegisterUrl(service.String())
+	data.CASLogin = GetCasLoginUrl(service.String())
+
+	fullname := strings.TrimSpace(c.FormValue("fullname"))
+	email := strings.ToLower(strings.TrimSpace(c.FormValue("email")))
+	password := c.FormValue("password")
+
+	api_v1_url_for_register_user := fmt.Sprintf("http://%s/api/v1/register/", *OSFHost)
+	jsonStr := fmt.Sprintf("{\"email1\": \"%s\", \"email2\": \"%s\", \"password\": \"%s\", \"fullName\": \"%s\"}", email, email, password, fullname)
```

Comment:
Use a raw string here ([defined with backticks](https://golang.org/ref/spec#String_literals)) so you can avoid all the escaping.